aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/SamplingRuleApplier.java

```diff
@@ -171,6 +230,71 @@ SamplingResult shouldSample(
         .build();
   }
 
+  long getNextSnapshotTimeNanos() {
+    return nextSnapshotTimeNanos;
+  }
+
+  SamplingRuleApplier withTarget(SamplingTargetDocument target, Date now) {
+    Sampler newFixedRateSampler = createFixedRate(target.getFixedRate());
+    Sampler newReservoirSampler = Sampler.alwaysOff();
+    long newReservoirEndTimeNanos = System.nanoTime();
+    // Not well documented but a quota should always come with a TTL
+    if (target.getReservoirQuota() != null && target.getReservoirQuotaTtl() != null) {
+      newReservoirSampler = createRateLimited(target.getReservoirQuota());
+      newReservoirEndTimeNanos =
+          clock.nanoTime()
+              + Duration.between(now.toInstant(), target.getReservoirQuotaTtl().toInstant())
```

Comment:
Is it typical that a TTL is actually an expiration timestamp and not a duration of time? If not, maybe worth referencing the API docs to clarify the TTL is in fact a timestamp to avoid confusion: https://docs.aws.amazon.com/xray/latest/api/API_SamplingTargetDocument.html

Reply:
This was indeed very confusing, linking seems good